xml/hwbp_registers.xml

```diff
@@ -3,9 +3,13 @@
     and Debug Mode to prevent untrusted user code from causing entry into Debug
     Mode without the OS's permission.
 
-    In this section XLEN means MXLEN when in M-mode, and DXLEN when in Debug Mode.
-    Note that this makes several of the fields in \RcsrTdataOne move around based
-    on the current execution mode and value of MXLEN.
+    In this section XLEN means MXLEN when in M-mode, and DXLEN when in Debug
+    Mode. On systems where XLEN is writable, this is handled as follows.
```

Comment:
"XLEN" isn't writable.  MXL and SXL and UXL and VSXL are possibly writable and affect XLEN in various modes.  You can, for instance, have programmable MXL which allows MXLEN to change.  But you can also have hard-wired values which only allow for MXLEN=64 and SXLEN=32.  So XLEN changes every time you drop into S mode without anything being programmable.

Since these registers are only accessible in M mode, perhaps only MXL matters for the purposes of software reads.  But also consider that dropping into 32b S mode might affect things like how many ASID bits match on tdata3 even though S mode can't see tdata3 directly.  So I think that the concern here is whether XLEN can change, not whether anything is writable.

Reply:
Good point that XLEN isn't writable.

I think we're OK otherwise, but not 100% sure everything is always right when XLEN changes.